akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -18,8 +18,10 @@ import akka.event.Logging.simpleName
  */
 private[akka] object StreamLayout {
 
-  // compile-time constant
-  final val Debug = true
+  // Turn this on to add run-time checks about Stream consistency during
+  // stream operations.
```

Comment:
it only validates during materialization, not during opertions (unless a stream operation does materialization of streams of course)